.github/workflows/ci.yml

```diff
@@ -1,32 +1,98 @@
 name: CI
 
-env:
-  COMPOSE_INTERACTIVE_NO_CLI: 1
-
 on:
   push:
-    branches: [ 3.x, 2.x, master ]
   pull_request:
     branches: [ 3.x, 2.x, master ]
 
 jobs:
   tests:
     runs-on: ubuntu-latest
+    container: shivammathur/node:latest
 
     strategy:
       matrix:
-        php: ["7.4", "8.0"]
-        laravel: ["^6.0", "^8.0"]
+        php: ['7.4', '8.0', '8.1']
+        laravel: ['^6.0', '^8.0']
+        exclude:
+          - php: '8.1'
+            laravel: '^6.0'
+
+    services:
+      postgres:
+        image: postgres:latest
+        env:
+          POSTGRES_PASSWORD: password
+          POSTGRES_USER: root
+          POSTGRES_DB: main
+        ports:
+          - 5432/tcp
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 3
+
+      mysql:
+        image: mysql:5.7
+        env:
+          MYSQL_ALLOW_EMPTY_PASSWORD: false
+          MYSQL_ROOT_PASSWORD: password
+          MYSQL_DATABASE: main
+        ports:
+          - 3306/tcp
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
```

Comment:
What are these pings used for?

Reply:
Checks the health of the services so that the "main action" only starts after all services are healthy.

Reply:
Makes sense